src/store/data.ts

```diff
@@ -260,12 +278,18 @@ export const gc = (data: InMemoryData) => {
       delete data.refCount[entityKey];
       data.records.base.delete(entityKey);
       data.gcBatch.delete(entityKey);
+      if (storage.current) {
+        persistanceBatch[prefixKey('r', entityKey)] = undefined;
+      }
 
       // Delete all the entity's links, but also update the reference count
       // for those links (which can lead to an unrolled recursive GC of the children)
       const linkNode = data.links.base.get(entityKey);
       if (linkNode !== undefined) {
         data.links.base.delete(entityKey);
+        if (storage.current) {
+          persistanceBatch[prefixKey('l', entityKey)] = undefined;
```

Comment:
We collect during gc, the write can be a microtask later but that doesn't really matter since then it's just a batch of removes and writes